c/tskit/trees.h

```diff
@@ -277,79 +277,101 @@ int tsk_treeseq_simplify(const tsk_treeseq_t *self, const tsk_id_t *samples,
 int tsk_treeseq_kc_distance(
     tsk_treeseq_t *self, tsk_treeseq_t *other, double lambda_, double *result);
 
-/* TODO do these belong in trees or stats? They should probably be in stats.
- * Keep them here for now until we figure out the correct interface.
- */
-int tsk_treeseq_genealogical_nearest_neighbours(tsk_treeseq_t *self, tsk_id_t *focal,
-    size_t num_focal, tsk_id_t **reference_sets, size_t *reference_set_size,
-    size_t num_reference_sets, tsk_flags_t options, double *ret_array);
-int tsk_treeseq_mean_descendants(tsk_treeseq_t *self, tsk_id_t **reference_sets,
-    size_t *reference_set_size, size_t num_reference_sets, tsk_flags_t options,
+int tsk_treeseq_genealogical_nearest_neighbours(const tsk_treeseq_t *self,
+    const tsk_id_t *focal, size_t num_focal, tsk_id_t *const *reference_sets,
+    const size_t *reference_set_size, size_t num_reference_sets, tsk_flags_t options,
     double *ret_array);
+int tsk_treeseq_mean_descendants(const tsk_treeseq_t *self,
+    tsk_id_t *const *reference_sets, const size_t *reference_set_size,
+    size_t num_reference_sets, tsk_flags_t options, double *ret_array);
 
 /* TODO change all these size_t's to tsk_size_t */
 
 typedef int general_stat_func_t(size_t K, double *X, size_t M, double *Y, void *params);
```

Comment:
Should this have ``const double *X`` and ``const void *params``?

Reply:
For reason I started replying about this in the other comment. The commit I just forced push now has this updated (and renamed the params to be more readable). But see the other comment for why I only consted one param and not params.